pkg/registry/generic/etcd/etcd.go

```diff
@@ -301,7 +301,7 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
 		}
 		if newVersion != version {
 			// TODO: return the most recent version to a client?
-			return nil, nil, kubeerr.NewConflict(e.EndpointName, name, fmt.Errorf("the resource was updated to %d", version))
+			return nil, nil, kubeerr.NewConflict(e.EndpointName, name, fmt.Errorf("another client made an update to this object in resource version %d", version))
```

Comment:
That's fair - hadn't followed the thought through to that.

"the resourceVersion you reported (%v) is out-of-date, the object has been modified; please re-read the object and try your change again"

How about: "the object has been modified; please apply your changes to the latest version and try again"

----- Original Message -----
> > @@ -301,7 +301,7 @@ func (e *Etcd) Update(ctx api.Context, obj
> > runtime.Object) (runtime.Object, bool
> >  		}
> >  		if newVersion != version {
> >  			// TODO: return the most recent version to a client?
> > -			return nil, nil, kubeerr.NewConflict(e.EndpointName, name,
> > fmt.Errorf("the resource was updated to %d", version))
> > +			return nil, nil, kubeerr.NewConflict(e.EndpointName, name,
> > fmt.Errorf("another client made an update to this object in resource
> > version %d", version))
> 
> Sure, I don't think that printing the old one or not is a big deal. I still
> think printing the new version basically enables user error; I can't think
> of a way to use that information safely.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/9569/files#r32166322